### DISCUSSION TITLE
How to get all messages from a Telegram chat?

### DISCUSSION THREAD

**Discussion by Evertt** [telegraf/telegraf#1822] [Q&A]
On [this docs page on core.telegram.org](https://core.telegram.org/method/messages.getMessages) it seems to suggest that there is a method available to get all the messages from a chat. At least, if I understand it correctly. To be honest, their code block looks very alien to me and I'm not quite sure how to interpret it.

But it even says:
> **Bots can use this method**

But I have not been able to find a method in Telegraf that provides this feature. Is it possible to get all the messages from a chat from Telegram? Because at the moment, I am saving all of those messages in my own database and it would save me so much storage if I could just get them from Telegram and not have to save them in my own database.

**Comment by MKRhere** [comment]
The page you're looking at is MTProto docs. Telegraf is a Bot API library, whose docs only exist at https://core.telegram.org/bots/api. `getMessages` is not a method available to the Bot API.
